contracts/message-based/PriceAware.sol

```diff
@@ -95,16 +86,18 @@ contract PriceAware is Ownable {
     address signer = hashWithPrefix.recover(signature);
     require(isSignerAuthorized(signer), "Signer not authorized");
 
-    //7. We extract timestamp from callData
+    // 7. We extract timestamp from callData
 
     uint256 dataTimestamp;
     assembly {
-      //Calldataload loads slots of 32 bytes
-      //The last 65 bytes are for signature + 1 for data size
-      //We load the previous 32 bytes
+      // Calldataload loads slots of 32 bytes
+      // The last 65 bytes are for signature + 1 for data size
+      // We load the previous 32 bytes
       dataTimestamp := calldataload(sub(calldatasize(), 98))
     }
 
+    // Data timestamp validation
```

Comment:
Timestamp checking logic should be moved to a separate method

Reply:
Done